python/salpytools/states.py

```diff
@@ -167,102 +86,135 @@ def wake(self):
 
 class StandbyState(DefaultState):
 
-    def __init__(self, device, model, events=('SummaryState',), tsleep=0.5):
-        super(StandbyState, self).__init__('STANDBY', device, model, events, tsleep)
+    def __init__(self, subsystem_tag, events=('SummaryState',), tsleep=0.5):
+        super(StandbyState, self).__init__('STANDBY', subsystem_tag, events, tsleep)
 
-    def exit(self):
-        self.model.change_state("OFFLINE")
+    def exit_control(self, model):
+        model.change_state("OFFLINE")
+        self.send_logEvent("SummaryState", SummaryStateValue=4)
 
-    def start(self):
-        self.model.change_state("DISABLED")
+    def start(self, model):
+        model.change_state("DISABLED")
+        self.send_logEvent("SummaryState", SummaryStateValue=1)
+
+    def exit(self, model):
+        print("Standby: exit() not implemented")
+
+    def do(self, model):
+        print("Standby: do() not implemented")
 
     def on_heartbeat(self, model):
         pass
 
 class DisabledState(DefaultState):
 
-    def __init__(self, device, model, events=('SummaryState',), tsleep=0.5):
-        super(DisabledState, self).__init__('DISABLED', device, model, events, tsleep)
+    def __init__(self, subsystem_tag, events=('SummaryState',), tsleep=0.5):
+        super(DisabledState, self).__init__('DISABLED', subsystem_tag, events, tsleep)
 
-    def enable(self):
-        self.model.change_state("ENABLED")
+    def enable(self, model):
+        model.change_state("ENABLED")
+        self.send_logEvent("SummaryState", SummaryStateValue=2)
 
-    def go_to_standby(self):
-        self.model.change_state("STANDBY")
+    def standby(self, model):
+        model.change_state("STANDBY")
+        self.send_logEvent("SummaryState", SummaryStateValue=5)
+
+    def exit(self, model):
+        print("Disabled: exit() not implemented")
+
+    def do(self, model):
+        print("Disabled: do() not implemented")
 
-    def on_heartbeat(self):
+    def on_heartbeat(self, model):
         pass
 
-    def on_incoming_messaging_error(self):
+    def on_incoming_messaging_error(self, model):
         pass
 
-    def on_interrupt_end_loop(self):
+    def on_interrupt_end_loop(self, model):
         pass
 
-    def on_interrupt_process_triggers(self):
+    def on_interrupt_process_triggers(self, model):
         pass
 
 class EnabledState(DefaultState):
 
-    def __init__(self, device, model, events=('SummaryState',), tsleep=0.5):
-        super(EnabledState, self).__init__('ENABLED', device, model, events, tsleep)
+    def __init__(self, subsystem_tag, events=('SummaryState',), tsleep=0.5):
+        super(EnabledState, self).__init__('ENABLED', subsystem_tag, events, tsleep)
+
+    def disable(self, model):
+        model.change_state("DISABLED")
+        self.send_logEvent("SummaryState", SummaryStateValue=1)
 
-    def disable(self):
-        self.model.change_state("DISABLED")
+    def exit(self, model):
+        print("Enabled: exit() not implemented")
```

Comment:
Please, use `self.log.debug` instead of `print`

Reply:
done